Explorer/Assets/Scripts/ECS/StreamableLoading/Tests/ConcludeTextureLoadingSystemShould.cs

```diff
@@ -22,7 +22,19 @@ protected override Entity CreateSuccessIntention()
         {
             var lr = new LoadingRequest { WebRequest = UnityWebRequestTexture.GetTexture(successPath) };
             lr.WebRequest.SendWebRequest();
-            return world.Create(new GetTextureIntention { CommonArguments = new CommonLoadingArguments(successPath) }, lr);
+
+            return world.Create(new GetTextureIntention
```

Comment:
Do you know if the heap allocations for each component are copied internally? Or can we pass a stack allocated struct, would this be faster or unnessecary?

Reply:
I didn't fully get the intention of your question, but I hope it will answer it:
1. When a new archetype arises chunks are preallocated based on each component's size: for ref component is just `IntPtr.Size`, and for value types, it's the size of the structure (that is known on compile-time by definition)
2. Having a structure facilitates data locality, omitting the necessity of additional addressing of a memory cell
3. Heap allocations are not copied (whatever you meant), just an address is stored

Reply:
>Oh this is interesting to me, my assumption was that specific components would be stored in contiguous memory structures to avoid cache misses. Hence if we pass a stack allocated component, it would be copied into the same memory region as all other same components (or archetypes?).

It is true but only for value types

Reply:
>In this case the cache misses would happen anyway, because the IntPtr points to random memory locations on the heap.

It is true for reference types